src/steps/fetchServices.ts

```diff
@@ -0,0 +1,128 @@
+import {
+  IntegrationStep,
+  IntegrationStepExecutionContext,
+  createIntegrationEntity,
+  createIntegrationRelationship,
+  JobState,
+} from '@jupiterone/integration-sdk';
+import _ from 'lodash';
+
+import { OnCall, Service } from '../types';
+import { requestAll } from '../pager-duty';
+import { reduceGroupById } from '../utils/reduceGroupById';
+
+const step: IntegrationStep = {
+  id: 'fetch-services',
+  name: 'Fetch Services',
+  dependsOn: ['fetch-teams', 'fetch-users'],
+  types: ['pagerduty_service'],
+  async executionHandler({
+    logger,
+    jobState,
+    instance,
+  }: IntegrationStepExecutionContext) {
+    logger.info('Requesting /services endpoint');
+    const { apiKey } = instance.config;
+    const services = await requestAll<Service>('/services', 'services', apiKey);
+
+    await buildServiceEntities(jobState, services);
+
+    await buildTeamRelations(jobState, services);
+
+    logger.info('Requesting /oncalls endpoint');
+    const oncalls = await requestAll<OnCall>('/oncalls', 'oncalls', apiKey);
+
+    await buildOnCallRelations(jobState, oncalls, services);
+  },
+};
+
+async function buildServiceEntities(
+  jobState: JobState,
+  services: Service[],
+): Promise<void> {
+  const serviceEntities = services.map((service) =>
+    createIntegrationEntity({
+      entityData: {
+        source: {
+          id: service.id,
+          name: service.name,
+          description: service.description,
+          summary: service.summary,
+        },
+        assign: {
+          _key: `service:${service.id}`,
+          _type: 'pagerduty_service',
+          _class: 'Service',
+          _rawData: [{ name: 'service', rawData: service }],
+          category: ['software'],
+          endpoints: [service.html_url],
+        },
+      },
+    }),
+  );
+  await jobState.addEntities(serviceEntities);
+}
+
+async function buildTeamRelations(
+  jobState: JobState,
+  services: Service[],
+): Promise<void> {
+  const teamServiceGrouping = reduceGroupById(services, 'teams');
+
+  await jobState.iterateEntities(
+    { _type: 'pagerduty_team' },
+    async (teamEntity) => {
+      const id = teamEntity._key.split(':')[1];
+      if (teamServiceGrouping[id]) {
+        const serviceRelationships = teamServiceGrouping[id].map((service) =>
+          createIntegrationRelationship({
+            _class: 'ASSIGNED',
+            fromKey: `service:${service.id}`,
+            fromType: `pagerduty_service`,
+            toKey: teamEntity._key,
+            toType: teamEntity._type,
+          }),
```

Comment:
There is a form of this which allows:
```
          createIntegrationRelationship({
            _class: 'ASSIGNED',
            from: service,
            to: teamEntity,
          }),
```
This would be ideal because it avoids knowledge here about how to build the keys, and what the type values are. However, we're going to need to be able to find entities by properties other than just `_type`. Also, the `teamEntity` should have a property other than `_key` which is used to identify the related entity. Anyway, created https://github.com/JupiterOne/integration-sdk/issues/79.